lib/puppet.rb

```diff
@@ -18,6 +18,8 @@
 require 'puppet/external/pson/common'
 require 'puppet/external/pson/version'
 require 'puppet/external/pson/pure'
+require 'gettext-setup'
+require 'locale'
```

Comment:
They're used by our [packaging](https://github.com/puppetlabs/packaging/blob/master/tasks/gem.rake#L65-L73) when generating artifacts. In the case of gems, when we package up a gem of puppet, the packaging reads this yaml file to determine what dependencies that gem should express. Our Gemfile also reads this file in to gather platform specific dependencies: https://github.com/puppetlabs/puppet/blob/master/Gemfile#L78-L99.

It's a bit overly complicated/duplicitous I think, but you know.. tech debt.